app/controllers/users_controller.rb

```diff
@@ -21,9 +30,50 @@ def create
     end
   end
 
+  def edit
+    find_user_by_id params[:id]
+  end
+
+  def update
+    find_user_by_id params[:id]
+    if @user.update user_params
+      flash[:success] = t "flash.success_user_updated"
+      redirect_to user_path @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    find_user_by_id(params[:id]).destroy
+    flash[:success] = t "flash.success_user_deleted"
+    redirect_to users_url
+  end
+
   private
 
+  def find_user_by_id id
+    @user = User.find_by id:
```

Comment:
aloo. show em có handle việc user not found mà edit ko có -> bổ sung cho edit => từ đso => lặp code 

=> em có thể find user và handle việc user not found ở method này luôn
sau đó dùng action_filter (cụ thẻ là before_action) để gọi cho edit, show., update, delete luôn hì 

```
def find_user_by_id
  @user = User.find_by params[:id]
   return if @user

   flash[:warning] = t "flash.warning_user_not_found"
   redirect_to users_path
end
```

```
before_action :find_user_by_id, only: %i(edit update show destroy)

def edit; end
```